File: Chapter 03/C#/Section_3_3_3_LSP/After/Forward.cs
using System;

namespace Chapter03.LSP.After;

/// <summary>
/// LSP SOLUTION: True Substitutability.
/// 
/// ARCHITECTURE NOTE: A Forward is a true substitute for a Player. It fully 
/// honors the contract set by the base class. It doesn't throw a 
/// NotImplementedException or refuse to do the work.
/// </summary>
public class Forward : Player 
{
    /// <summary>
    /// Flawlessly fulfills the base class contract.
    /// </summary>
    public override void PlayFieldPosition() 
    {
        Console.WriteLine("  [Forward] Leading the attack and trying to score."); 
    } 
}

File: Chapter 03/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter03/section_3_3_3_lsp/after/Forward.java
package com.grokkingsoftwarearchitecture.chapter03.section_3_3_3_lsp.after;

/**
 * LSP SOLUTION: True Substitutability.
 * * ARCHITECTURE NOTE: A Forward is a true substitute for a Player. It fully 
 * honors the contract set by the base class. It doesn't throw an 
 * UnsupportedOperationException or refuse to do the work.
 */
public class Forward extends Player {
    
    /**
     * Flawlessly fulfills the base class contract.
     */
    @Override
    public void playFieldPosition() {
        System.out.println("  [Forward] Leading the attack and trying to score.");
    }
}